services/merak-topo/handler/handler.go

```diff
@@ -346,33 +345,45 @@ func UpdateComputenodeInfo(k8client *kubernetes.Clientset, topo_id string) error
 
 	elaps := time.Since(start_time0)
 
-	log.Printf("updatecomputenode:=========complete updating compute nodes info in DB =========== %v", elaps)
+	utils.Logger.Info("Complete", "updating compute nodes info in DB (in second) ", elaps)
 
-	return nil
+	if err_flag == 1 {
+		err_return = errs
+		err_flag = 0
+	} else {
+		err_return = nil
+	}
+	return err_return
 }
 
-func Info(k8client *kubernetes.Clientset, topo_id string, returnMessage *pb.ReturnTopologyMessage) error {
+func Info(k8client *kubernetes.Clientset, topo_id string, returnMessage *pb.ReturnTopologyMessage, topoPrefix string, namespace string) error {
+
+	err_flag := 0
+	errs := errors.New("fails to handle request CHECK")
+	var err_return error
 
-	topo, err := database.FindTopoEntity(topo_id, "")
+	topo, err := database.FindTopoEntity(topoPrefix, "")
 
 	if err != nil {
-		return fmt.Errorf("update compute node info: query topology_id error %s", err)
+		utils.Logger.Warn("request CHECK", topoPrefix, err.Error())
 	}
 
 	k8s_nodes, err1 := k8client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 
 	if err1 != nil {
-		return fmt.Errorf("INFO: list k8s host nodes info error %s", err1)
+		utils.Logger.Error("k8s cluster no response ", "corev1", err1.Error())
+		err_flag = 1
```

Comment:
@zzengcs Same here!!

Reply:
Yes, it's the same. All similar log msg now are changed 
1) for all k8s env query/check---- return err when fails
2) for all merak-topo DB query/check -- use error flag to return err at end of function. 

@cj-chung , please give some comments if the design above is not appropriate. Thanks!